src/IO/WriteBufferFromS3.h

```diff
@@ -33,31 +33,48 @@ namespace DB
  */
 class WriteBufferFromS3 : public BufferWithOwnMemory<WriteBuffer>
 {
+public:
+    class IExecutor
```

Comment:
You don't really need this interface.

Reply:
It was needed to resolve dependency cycle, in 7531e55703c1b1b88de69ec9cf1f3c50e701d49c .